lib/rules/no-assignment-of-computed-property-dependencies.js

```diff
@@ -0,0 +1,191 @@
+ *
+ * @param {Node} nodeClass - Node for the Ember class
+ * @returns {String[]} - list of tracked properties used inside the class
+ */
+function findTrackedProperties(nodeClassDeclaration) {
+  return nodeClassDeclaration.body.body
+    .filter(
+      (node) =>
+        types.isClassProperty(node) &&
+        decoratorUtils.hasDecorator(node, 'tracked') &&
+        types.isIdentifier(node.key)
+    )
+    .map((node) => node.key.name);
+}
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'disallow assignment of untracked properties that are used as computed property dependencies',
+      category: 'Computed Properties',
+      recommended: false,
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-assignment-of-computed-property-dependencies.md',
+    },
+    fixable: 'code',
+    schema: [],
+  },
+
+  ERROR_MESSAGE,
+
+  create(context) {
+    if (emberUtils.isTestFile(context.getFilename())) {
+      // This rule does not apply to test files.
+      return {};
+    }
+
+    // State being tracked for this file.
+    let trackedProperties = new Set();
+    let computedPropertyDependentKeys = new Set();
+    let currentEmberModule = null;
+
+    return {
+      ClassDeclaration(node) {
+        if (emberUtils.isAnyEmberModule(context, node)) {
+          currentEmberModule = node;
+
+          // Gather computed property dependent keys from this class.
+          computedPropertyDependentKeys = new Set(findComputedPropertyDependentKeys(node));
+
+          // Gather tracked properties from this class.
+          trackedProperties = new Set(findTrackedProperties(node));
+        }
+      },
+
+      CallExpression(node) {
+        if (emberUtils.isAnyEmberModule(context, node)) {
+          currentEmberModule = node;
+
+          // Gather computed property dependent keys from this class.
+          computedPropertyDependentKeys = new Set(findComputedPropertyDependentKeys(node));
+        }
+      },
+
+      'ClassDeclaration:exit'(node) {
+        if (currentEmberModule === node) {
+          // Leaving current class.
+          currentEmberModule = null;
+        }
+      },
+
+      'CallExpression:exit'(node) {
+        if (currentEmberModule === node) {
+          // Leaving current class.
+          currentEmberModule = null;
+        }
+      },
+
+      AssignmentExpression(node) {
+        if (!currentEmberModule) {
+          // Not inside an Ember class.
+          return;
+        }
+
+        // Ensure this is an assignment with `this.x = ` or `this.x.y = `.
+        if (!propertySetterUtils.isThisSet(node)) {
+          return;
+        }
+
+        const sourceCode = context.getSourceCode();
+        const nodeTextLeft = sourceCode.getText(node.left);
+        const nodeTextRight = sourceCode.getText(node.right);
+        const propertyName = nodeTextLeft.replace('this.', '');
+
+        if (propertyName.startsWith('args.')) {
```

Comment:
We should make sure that we're in a Glimmer component for this check. You could have an `args` property that needs to be tracked on a class Ember component, or in a service.

Reply:
Fixed.